### PR TITLE
Add gulp option

### PR DESCRIPTION
**Pull request by jimmynotjim**
This PR adds the option to start a project with Gulp instead of Grunt. It uses pretty much the same exact setup, just with a few tweaks.

## Additions

- Add Gulp templates to a `/gulp` directory
  - Tasks are broken into individual files
  - All tasks use a central config to set up directory structures

## Removals

- None

## Changes

- Adds `askForBuildTool` option during generation to allow user to choose their build tool
- Moves Grunt template files to a `/grunt` directory

## Testing

- `git checkout gulp option && npm link`
- create dummy project, outside this repo, `mk test-project && cd_`
- run the generator in test-project `yo cf` making sure to choose Gulp as your build tool
- after NPM & Bower complete, run `gulp build`
- PROFIT!!!

## Review

- @contolini 
- @ascott1 
- @KimberlyMunoz 
- @sebworks 
- Anyone else interested in Gulp???

## Screenshots

screen shot 2015-10-06 at 8 28 02 pm

screen shot 2015-10-06 at 8 28 40 pm

## Todos

- None

## Checklist

* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices 
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

File: app/templates/_bower.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
   },
   "license": "<%= props.license %>",
   "dependencies": { <% for(var i=0; i<components.length; i++) { %>
-    "<%= components[i].name %>": "<%= components[i].ver%>"<% if (i < components.length - 1) { %>,<% } %><% } %>
+    "<%= components[i].name %>": "<%= components[i].ver%>"<% if (i < components.length - 1) { %>,<% } %><% } %>,
+    "jquery.easing": "~1.3.0"
```
Comment (contolini) [inline app/templates/_bower.json]: Why was jquery.easing added?

Comment (wpears) [inline app/templates/_bower.json]: Probably for `https://github.com/jimmynotjim/generator-cf/blob/gulp-option/app/templates/gulp/config/browserify-shims.js`

**Comment by contolini** [comment]
@jimmynotjim This is really exciting! Everything builds fine for me. I filed a PR against your fork with some smoke tests.

File: app/templates/_bower.json (hunk 1 above)
Comment (jimmynotjim) [inline app/templates/_bower.json]: Whoops, good catch, I thought I'd removed everything browserify related.

**Comment by jimmynotjim** [comment]
Merged smoke tests and looking good.

**Comment by contolini** [comment]
:+1: S'all good here.

**Comment by jimmynotjim** [comment]
Let me test this just once more before we merge it.

File: app/templates/gulp/gulp/config.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        loc.lib,
+        loc.lib + '/cf-typography/src'
+      ],
+      compress: true
+    }
+  },
+  scripts: {
+    src: [
+      loc.lib + '/jquery/dist/jquery.js',
+      loc.lib + '/jquery.easing/js/jquery.easing.js',
```
Comment (anselmbradford) [inline app/templates/gulp/gulp/config.js]: Remove easing here if removed in https://github.com/cfpb/generator-cf/pull/103#discussion_r41351768

File: app/templates/gulp/gulp/config.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      ],
+      dest: loc.dist
+    },
+    icons: {
+      src:  loc.lib + '/cf-icons/src/fonts/*',
+      dest: loc.dist + '/static/fonts/'
+    },
+    vendorjs: {
+      src: [
+        loc.lib + '/box-sizing-polyfill/boxsizing.htc',
```
Comment (anselmbradford) [inline app/templates/gulp/gulp/config.js]: This is only needed if cf-grid is included. Just flagging if that for some reason wouldn't be needed.

File: app/templates/gulp/gulp/tasks/copy.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+var gulp = require( 'gulp' );
+var $ = require( 'gulp-load-plugins' )();
```
Comment (anselmbradford) [inline app/templates/gulp/gulp/tasks/copy.js]: https://github.com/cfpb/cfgov-refresh/issues/830 could be updated here or later when 830 is closed.

File: app/templates/gulp/gulp/tasks/lint.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+ * Lints the gulpfile for errors
+ */
+gulp.task( 'lint:build', function() {
+  return gulp.src( config.gulp )
+    .pipe( $.eslint() )
+    .pipe( $.eslint.format() )
+    .on( 'error', handleErrors );
+} );
+
+
```
Comment (anselmbradford) [inline app/templates/gulp/gulp/tasks/lint.js]: Extra newline

File: app/templates/gulp/gulp/tasks/styles.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    } ) );
+} );
+
+gulp.task( 'styles:ie', function() {
+  return gulp.src( config.cwd + config.src )
+    .pipe( $.less( config.settings ) )
+    .on( 'error', handleErrors )
+    .pipe( $.replace(
+      /url\('chosen-sprite.png'\)/ig,
+      'url("/static/img/chosen-sprite.png")'
```
Comment (anselmbradford) [inline app/templates/gulp/gulp/tasks/styles.js]: Should chosen be included?

File: app/templates/gulp/gulp/config.js (hunk 2 above)
Comment (jimmynotjim) [inline app/templates/gulp/gulp/config.js]: Oh, now I'm not sure, cause it was in [the Gruntfile](https://github.com/jimmynotjim/generator-cf/blob/gulp-option/app/templates/grunt/_Gruntfile.js#L46)

File: app/templates/gulp/gulp/config.js (hunk 3 above)
Comment (jimmynotjim) [inline app/templates/gulp/gulp/config.js]: Copied from [the Gruntfile](https://github.com/jimmynotjim/generator-cf/blob/gulp-option/app/templates/grunt/_Gruntfile.js#L230)

File: app/templates/gulp/gulp/tasks/styles.js (hunk 6 above)
Comment (jimmynotjim) [inline app/templates/gulp/gulp/tasks/styles.js]: Probably not

File: app/templates/gulp/gulp/tasks/lint.js (hunk 5 above)
Comment (jimmynotjim) [inline app/templates/gulp/gulp/tasks/lint.js]: Fixed

**Comment by contolini** [comment]
Good-to-go?

Let's hope so.

Mergin'.

**Comment by ascott1** [comment]
YASSSSS

**Pull request merged**